litellm/llms/databricks/client.py

```diff
@@ -0,0 +1,514 @@
+                    "In order to make asynchronous calls,"
+                    " Databricks API base URL and API key must both be set."
+                ),
+            )
+        elif not databricks_sdk_installed:
+            raise DatabricksError(
+                status_code=400,
+                message=(
+                    "If Databricks API base URL and API key are not provided,"
+                    " the databricks-sdk Python library must be installed."
+                ),
+            )
+
+    if support_async and api_base is not None and api_key is not None:
+        async_http_handler = (
+            http_handler
+            if isinstance(http_handler, AsyncHTTPHandler)
+            else AsyncHTTPHandler(timeout=timeout)
+        )
+        return DatabricksModelServingAsyncHTTPHandlerWrapper(
+            api_base=api_base,
+            api_key=api_key,
+            http_handler=async_http_handler,
+            custom_llm_provider=custom_llm_provider,
+            logging_obj=logging_obj,
+            custom_endpoint=custom_endpoint,
+            headers=headers,
+        )
+    elif http_handler is not None and api_base is not None and api_key is not None:
+        http_handler = (
+            http_handler
+            if isinstance(http_handler, HTTPHandler)
+            else HTTPHandler(timeout=timeout)
+        )
+        return DatabricksModelServingHTTPHandlerWrapper(
+            api_base=api_base,
+            api_key=api_key,
+            http_handler=http_handler,
+            custom_llm_provider=custom_llm_provider,
+            logging_obj=logging_obj,
+            custom_endpoint=custom_endpoint,
+            headers=headers,
+            streaming_decoder=streaming_decoder,
+        )
+    else:
+        return DatabricksModelServingWorkspaceClientWrapper(
+            logging_obj=logging_obj, api_base=api_base, api_key=api_key, headers=headers
+        )
+
+
+class DatabricksModelServingWorkspaceClientWrapper(DatabricksModelServingClientWrapper):
+    """
+    A wrapper around the Databricks Model Serving API client that uses the Databricks SDK to
+    make inference calls. This wrapper is used when the Databricks API base URL and API key are not
+    provided, and the Databricks SDK is used to automatically retrieve the API key and base URL
+    from the current environment.
+
+    The Databricks SDK does not support asynchronous calls, so this wrapper is only used for
+    synchronous calls.
+
+    TODO: Support asynchronous execution with the Databricks SDK
+    """
+
+    def __init__(
+        self,
+        logging_obj: LiteLLMLoggingObj,
+        api_base: Optional[str],
+        api_key: Optional[str],
+        headers: Optional[Dict[Any, Any]],
+    ):
+        from databricks.sdk import WorkspaceClient
+
+        self.logging_obj = logging_obj
+        # NB: WorkspaceClient is not typed correctly in the Databricks SDK
+        self.client = WorkspaceClient(host=api_base, token=api_key)  # type: ignore
```

Comment:
instead of relying on the workspace client to also make the calls, 

Can we just use it to get the credentials in the environment? Similar to how boto3 and vertex ai's sdk's are used 


https://github.com/BerriAI/litellm/blob/3a5039e284adac1279964849879619fe2cef309c/litellm/llms/bedrock/chat/converse_handler.py#L249

Reply:
@krrishdholakia Unfortunately, this is not feasible with the Databricks SDK (https://github.com/BerriAI/litellm/pull/5717#issuecomment-2354012192)

Reply:
Oh - why ?